app/services/versioned_files_service/stacks_link_action.rb

```diff
@@ -11,7 +11,15 @@ def initialize(object:, version:)
     def call
       FileUtils.mkdir_p(stacks_object_path)
       shelve_file_map.each do |filename, md5|
-        LinkSupport.link(content_path_for(md5:), stacks_object_path.join(filename))
+        file_path = stacks_object_path.join(filename)
+
+        if file_path.to_s.starts_with?(object_path.to_s) || !file_path.to_s.starts_with?(stacks_object_path.to_s)
+          Honeybadger.notify("Skipping #{filename} because it would conflict with the AWFL directory layout or is otherwise outside the object directory")
```

Comment:
Since it is likely that our future selves wouldn't remember what AWFL is, it is referred to as "versioned layout" in the code. In this case, I'd suggest something like "it would conflict with the versioned object directory".